main/com.tunein/device.js

```diff
@@ -0,0 +1,102 @@
+                        } else if (content.find(item => item.text.toLowerCase() == 'stations')) {
+                            return resolve(
+                                content.filter(item => item.text.toLowerCase() == 'stations')
+                                       .flatMap(({children}) => children)
+                                       .filter(channel => (channel.type.toLowerCase() == 'audio' && channel.item.toLowerCase() == 'station'))
+                            );
+                        } else {
+                            return resolve(content.filter(channel => (channel.type.toLowerCase() == 'audio' && channel.item.toLowerCase() == 'station')));
+                        }; 
+                    } else {
+                        reject(response.data);
+                        return new Error(`HTTP request error: ${response.data.head.fault}`);
+                    }
+                })
+                .catch(err => throwError("streaming_service_unaccessible"));
+        }).catch(err => throwError("streaming_service_unaccessible"));
+    }
+
+    _get_channel_info(ch_list) {
+        return ch_list.map(
+            (x) => (({text, subtext='', current_track='', now_playing_id='', URL='', formats='', image=''}) => ({text, subtext, current_track, now_playing_id, URL, formats, image}))(x)
+            );
+    }
+
+    async get_search_channel({query, count = 0}) {
+        let req = {};
+        req.params = {};
+        req.url = SEARCH_PARAM;
+        req.params.query = query;
+        const available_channels = await this._get_tunein_response(req);
+        if ([undefined, null, 0].includes(count))
+            return this._get_channel_info(available_channels);
+        else
+            return this._get_channel_info(available_channels.slice(0,count));
+    }
+
+    async get_list_local_channels({count=0}) {
+        let req = {};
+        req.params = {};
+        req.url = BROWSE_PARAM;
+        req.params.c = 'local';
+        const available_channels = await this._get_tunein_response(req);
+        if ([undefined, null, 0].includes(count))
+            return this._get_channel_info(available_channels);
+        else
+            return this._get_channel_info(available_channels.slice(0,count));
+    }
+
+    async get_list_trending_channels({count = 0}) {
+        let req = {};
+        req.params = {};
+        req.url = BROWSE_PARAM;
+        req.params.c = 'trending';
+        const available_channels = await this._get_tunein_response(req);
+        if ([undefined, null, 0].includes(count))
+            return this._get_channel_info(available_channels);
+        else
+            return this._get_channel_info(available_channels.slice(0,count));
+    }
+
+    async do_play({query}) {
+        let req = {};
+        req.params = {};
+        req.url = SEARCH_PARAM;
+        req.params.query = query;
+        const available_channels = await this._get_tunein_response(req);
+        return available_channels[0].URL;
```

Comment:
In here you'll probably want to actually play, right?
The best way is to use the audio-player platform capability: https://almond-static.stanford.edu/jsdoc/thingpedia/latest/interfaces/capabilities.audioplayerapi.html which you access with `this.platform.getCapability('audio-player')` (and a null check)

The alternative is that you return an "audio" formatted object and let the default Genie code handle playing (This is what smartnews does). If you do that though, you will have limited control on pause from inside the skill.